packages/astro/src/core/render-context.ts

```diff
@@ -163,6 +164,11 @@ export class RenderContext {
 							streaming,
 							this.routeData
 						);
+
+						const responseCookies = getFromResponse(response);
+						if (result.cookies && responseCookies) {
+							result.cookies?.merge(responseCookies);
```

Comment:
```suggestion
							result.cookies.merge(responseCookies);
```

Not needed I think, due to the previous check